src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
+            fraudProofs: proofs,
+            onChainSlashes,
+            onChainLatestJoinChannelBlockHash,
+            outputStateSnapshotHash: disputeOutputStateSnapshotHash,
+            exitChannelBlocks,
+            disputeAuditingDataHash,
+            disputer: this.signerAddress,
+            disputeIndex: disputeIndex,
+            previousRecursiveDisputeIndex: ethers.MaxUint256,
+            timeout: timeoutDispute,
+            selfRemoval: false
+        };
+        return { dispute, disputeAuditingData };
     }
 
-    private async createNewDispute(
-        forkCnt: BigNumberish,
-        foldedParticipant: AddressLike,
-        foldedTransactionCnt: BigNumberish,
-        proofs: ProofStruct[]
+    public async createNewDispute(
+        forkCnt: number,
+        transactionCnt: number,
+        proofs: ProofStruct[],
+        timestamp: BigNumberish,
+        timeout?: TimeoutStruct
     ): Promise<void> {
-        const {
-            encodedLatestFinalizedState,
-            encodedLatestCorrectState,
-            virtualVotingBlocks
-        } = this.agreementManager.getFinalizedAndLatestWithVotes(
+        const { dispute, disputeAuditingData } = await this.createDisputeStruct(
             forkCnt,
-            this.signerAddress
+            transactionCnt,
+            proofs,
+            timestamp,
+            timeout
         );
 
-        this.p2pEventHooks.onInitiatingDispute?.();
         await retry(
             async () => {
                 const txResponse =
                     await this.stateChannelManagerContract.createDispute(
-                        this.channelId,
-                        forkCnt,
-                        encodedLatestFinalizedState,
-                        encodedLatestCorrectState,
-                        virtualVotingBlocks,
-                        foldedParticipant,
-                        foldedTransactionCnt,
-                        proofs,
-                        { gasLimit: 4000000 } //TODO! - gas limit
+                        dispute
+                    );
+                const txReceipt = await txResponse.wait();
+                return txReceipt;
+            },
+            {
+                maxRetries: 1, // Current implementation retries once
+                onRetry: (attempt, error) => {
+                    console.log("ERROR - DISPUTE CATCH ##", error);
+                    console.log(
+                        `Retrying dispute creation, attempt ${attempt}`
+                    );
+                }
+            }
+        );
+
+        // gossip to all peers
+    }
+
+    public async auditDispute(
+        dispute: DisputeStruct,
+        disputeAuditingData: DisputeAuditingDataStruct
+    ): Promise<AddressLike[]> {
+        let slashParticipants: AddressLike[] = [];
+        await retry(
+            async () => {
+                slashParticipants =
+                    (await this.stateChannelManagerContract.auditDispute(
```

Comment:
same as other comments - it returns TransactionResponse and auditing is done in local EVM